enso/experiment/finetuning.py

```diff
@@ -0,0 +1,87 @@
+import os
+import json
+
+import pandas as pd
+
+from indicoio.custom import Collection
+from finetune import Classifier, SequenceLabeler
+
+from enso.experiment import ClassificationExperiment
+from enso.config import RESULTS_DIRECTORY
+from enso.registry import Registry, ModeKeys
+
+
+@Registry.register_experiment(ModeKeys.CLASSIFY, requirements=[("Featurizer", "PlainTextFeaturizer")])
+class Finetune(ClassificationExperiment):
+    """
+    LanguageModel finetuning as an alternative to simple models trained on top of pretrained features.
+    """
+
+    param_grid = {}
+
+    def __init__(self, *args, **kwargs):
+        """Initialize internal classifier."""
+        super().__init__(*args, **kwargs)
+        self.model = Classifier(autosave_path=os.path.join(RESULTS_DIRECTORY, '.autosave'))
+
+    def fit(self, X, y):
+        """
+        :param X: `np.ndarray` of raw text sampled from training data.
+        :param y: `np.ndarray` of corresponding targets sampled from training data.
+        """
+        self.model.fit(X, y)
+
+    def predict(self, X, **kwargs):
+        """Predict results on test set based on current internal model."""
+        preds = self.model.predict_proba(X)
+        return pd.DataFrame.from_records(preds)
+
+
+@Registry.register_experiment(ModeKeys.SEQUENCE, requirements=[("Featurizer", "PlainTextFeaturizer")])
+class FinetuneSequenceLabel(ClassificationExperiment):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.model = SequenceLabeler(autosave_path=os.path.join(RESULTS_DIRECTORY, '.autosave'), val_size=0)
+
+    def fit(self, X, y):
+        self.model.fit(X, y)
+
+    def predict(self, X, **kwargs):
+        return self.model.predict(X)
+    
+
+@Registry.register_experiment(ModeKeys.SEQUENCE, requirements=[("Featurizer", "PlainTextFeaturizer")])
+class IndicoSequenceLabel(ClassificationExperiment):
+
+    def __new__(cls, *args, **kwargs):
+        raise Exception("DO NOT run this at the moment.... - waiting to hear from Madison")
```

Comment:
Nice.

Reply:
I can just totally remove this if you want?